test/Microsoft.AspNetCore.Mvc.Core.Test/ApplicationParts/ApplicationPartManagerTest.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Core.Test.ApplicationParts
+{
+    public class ApplicationPartManagerTest
+    {
+
+        [Fact]
+        public void PopulateFeature_InvokesAllProvidersSequentially_ForAGivenFeature()
+        {
+            // Arrange
+            var manager = new ApplicationPartManager();
+            manager.ApplicationParts.Add(new TestPart("A"));
+            manager.ApplicationParts.Add(new OtherPartType("B"));
+            manager.ApplicationParts.Add(new TestPart("C"));
+            manager.Providers.Add(new TestFeatureProvider((f, v) => f.Values.Add($"P1{v}")));
+            manager.Providers.Add(new TestFeatureProvider((f, v) => f.Values.Add($"P2{v}")));
+
+            var feature = new TestFeature();
+            
+            // Act
+            manager.PopulateFeature(feature);
+
+            // Assert
+            Assert.Equal(new[] { "P1A", "P1C", "P2A", "P2C" }, feature.Values.ToArray());
```

Comment:
BTW even though it's much easier to follow, I did have to read it a couple of times to understand why P1B/P2B don't show up. Can anything be done to the names of the test types to indicate that `TestPart` is of "interest" and that `OtherPartType` is "not of interest"?